spring-integration-core/src/main/java/org/springframework/integration/json/JsonPropertyAccessor.java

```diff
@@ -117,6 +120,22 @@ private static Integer maybeIndex(String name) {
 		}
 	}
 
+	/**
+	 * Check if the string is a numeric representation (all digits) or not.
+	 */
+	private static boolean isNumeric(String str) {
+		if (str == null || str.isEmpty()) {
+			return false;
+		}
+		final int length = str.length();
```

Comment:
We don't need to extract this variable: can simply be done inside `for()` expression.

Reply:
OK. Found another way to improve its performance:
```

	@Test
	void stringIterationPerformance() {
		int tmp = 0;
		String s = new String(new byte[64*1024]);

		{
			long st = System.nanoTime();
			int n = s.length();
			for(int i = 0; i < n; i++) {
				tmp += s.charAt(i);
			}
			st = System.nanoTime() - st;
			System.out.println("1 " + st);
		}

		{
			long st = System.nanoTime();
			for(char c : s.toCharArray()) {
				tmp += c;
			}
			st = System.nanoTime() - st;
			System.out.println("2 " + st);
		}
	}
```
The result is:
```
1 1680300
2 807000
```

So, `for(char c : s.toCharArray()) {` is twice faster then `for(int i = 0; i < n; i++) {`

Reply:
I'm curious in your results :)

Even if we iterate string using toCharArray first, it will create an additional memory allocation in size of the whole string every time. But in many cases in this method we will break at the first character and return. We are not trying to sum characters, we are looking for the first non-digit. On practice the first character in JSON key here is usually either digit (and the rest are as well) or not (and we return immediately). So this performance optimisation (if it does optimise something) will be relevant only for all digit strings, while significantly harming the performance of non-digit keys check.

But even if you want to test the performance of your alternative, I think the proper way would be to run it at least with something like JMH. For example:
```
@BenchmarkMode(Mode.AverageTime)
@OutputTimeUnit(TimeUnit.MILLISECONDS)
@State(Scope.Benchmark)
@Fork(value = 1)
@Warmup(iterations = 3, time = 1)
@Measurement(iterations = 5, time = 1)
public class StringIterationBenchmark {

    private static final String s = new String(new byte[64*1024]);


    public static void main(String[] args) throws RunnerException {
        Options opt = new OptionsBuilder()
            .include(StringIterationBenchmark.class.getSimpleName())
            .build();

        new Runner(opt).run();
    }

    @Benchmark
    public int iterateWithCharAt() {
        int tmp = 0;

        int n = s.length();
        for(int i = 0; i < n; i++) {
            tmp += s.charAt(i);
        }

        return tmp;
    }

    @Benchmark
    public int iterateWithToCharArray() {
        int tmp = 0;

        for(char c : s.toCharArray()) {
            tmp += c;
        }

        return tmp;
    }

}
```

Which on my machine (M1 MacBook Pro) doesn't look like an improvement:
```
Benchmark                                        Mode  Cnt  Score    Error  Units
StringIterationBenchmark.iterateWithCharAt       avgt    5  0.021 ±  0.001  ms/op
StringIterationBenchmark.iterateWithToCharArray  avgt    5  0.032 ±  0.001  ms/op
```

Reply:
OK. Don't have any objections any more.
I'm not familiar with JMH: will learn it other day.

For now I'm accepting your fix as is and back-porting it respectively to all supported versions.

Thank you for education!